packages/components/pie-notification/src/index.ts

```diff
@@ -1,24 +1,241 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-
-import { defineCustomElement } from '@justeattakeaway/pie-webc-core';
+import {
+    unsafeCSS,
+    nothing,
+    LitElement,
+    type TemplateResult,
+    type PropertyValues,
+} from 'lit';
+import { type StaticValue, html, unsafeStatic } from 'lit/static-html.js';
+import { defineCustomElement, validPropertyValues } from '@justeattakeaway/pie-webc-core';
+import { property, queryAssignedElements, state } from 'lit/decorators.js';
+import { type NotificationProps, variants, headingLevels } from './defs';
 import styles from './notification.scss?inline';
-import { NotificationProps } from './defs';
+
+import '@justeattakeaway/pie-icon-button';
+import '@justeattakeaway/pie-icons-webc/IconClose';
+import '@justeattakeaway/pie-icons-webc/IconInfoCircle';
+import '@justeattakeaway/pie-icons-webc/IconAlertCircle';
+import '@justeattakeaway/pie-icons-webc/IconAlertTriangle';
+import '@justeattakeaway/pie-icons-webc/IconCheckCircle';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-notification';
+const componentClass = 'c-notification';
+
+type InternalVariantType = Exclude<NotificationProps['variant'], undefined>;
```

Comment:
I think the built-in [`NonNullable`](https://www.typescriptlang.org/docs/handbook/utility-types.html#nonnullabletype) type might be slightly preferable, and you can probably use it inline rather than declaring it outside the class

Reply:
I was trying to avoid repeating `NonNullable<NotificationProps['variant']>` several times but I believe is better than declaring outside the class.